src/common/vector_operations/vector_cast.cpp

```diff
@@ -168,10 +168,126 @@ static bool NumericCastSwitch(Vector &source, Vector &result, idx_t count, strin
 	}
 }
 
+template <class T>
+bool FillEnum(string_t *source_data, ValidityMask &source_mask, const LogicalType &source_type, T *result_data,
+              ValidityMask &result_mask, const LogicalType &result_type, idx_t count, string *error_message) {
+	bool all_converted = true;
+	if (source_mask.AllValid()) {
+		for (idx_t i = 0; i < count; i++) {
+			auto string_value = source_data[i].GetString();
+			auto pos = EnumType::GetPos(result_type, string_value);
+			if (pos == -1) {
+				if (!error_message) {
+					throw CastException(source_type, result_type);
+				} else {
+					all_converted = false;
+					result_mask.SetInvalid(i);
+				}
+			} else {
+				result_data[i] = pos;
+			}
+		}
+	} else {
+		for (idx_t i = 0; i < count; i++) {
+			if (source_mask.RowIsValid(i)) {
```

Comment:
Since we are constructing a string and doing a lookup in an unordered_map, I think the code duplication for the no-null case is not required/will not make a significant difference.